code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -364,11 +364,17 @@
 /proc/hasHUD(mob/M as mob, hudtype)
 	if(istype(M, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = M
+		var/eye_hud
+		var/obj/item/clothing/glasses/G = H.glasses
+		// Checks for eye sensor HUD
+		var/obj/item/organ/internal/augment/eye_sensors/E = locate() in H.internal_organs
+		if(E)
+			eye_hud = E.selected_hud
 		switch(hudtype)
 			if("security")
-				return istype(H.glasses, /obj/item/clothing/glasses/hud/security) || istype(H.glasses, /obj/item/clothing/glasses/sunglasses/sechud)
+				return istype(G, /obj/item/clothing/glasses/hud/security) && G.active || istype(G, /obj/item/clothing/glasses/sunglasses/sechud) && G.active || istype(G, /obj/item/clothing/glasses/eyepatch/hud/security) && G.active || eye_hud == "Security"
```

Comment:
a better way to do this imho would honestly be to have a proc on glasses:
```
/obj/item/clothing/glasses/proc/is_security_hud()
	return FALSE
```
override this as needed on instances of glasses that count as security:
```
/obj/item/clothing/glasses/hud/security/is_security_hud()
	return TRUE && active
```
and so on for medical as well